src/shared_modules/utils/opensslWrapper.hpp

```diff
@@ -24,41 +25,130 @@ constexpr int RSA_PRIVATE {0};
 constexpr int RSA_PUBLIC {1};
 constexpr int RSA_CERT {2};
 
-namespace Utils
+template<typename T = OpenSSLPrimitives, typename U = OSPrimitives>
+class OpenSSL final
+    : public T
+    , public U
 {
+public:
+    explicit OpenSSL() {}
+    virtual ~OpenSSL() {}
+
+    /**
+     * Encrypts the input vector with the provided key
+     *
+     * @param filePath  The path to the file key string to encrypt the value
+     * @param input     The entry to be encrypted
+     * @param output    The resulting encrypted value
+     * @param cert      If the public key is in a certificate
+     * @return          The size of the encrypted output, -1 if error
+     */
+    int rsaEncrypt(const std::string& filePath, const std::string& input, std::string& output, bool cert = false)
+    {
+        RSA* rsa = nullptr;
+
+        createRSA(rsa, filePath, cert ? RSA_CERT : RSA_PUBLIC);
+
+        // Allocate memory for the encryptedValue
+        unsigned char* encryptedValue = (unsigned char*)malloc(T::RSA_size(rsa));
```

Comment:
Why are we using the old implementation? The previous function was different and did not use malloc.

```cpp
    int rsaEncrypt(const std::string& filePath, const std::string& input, std::string& output, bool cert = false)
    {
        RSA* rsa = nullptr;

        createRSA(rsa, filePath, cert ? RSA_CERT : RSA_PUBLIC);

        // Allocate memory for the encryptedValue
        std::vector<unsigned char> encryptedValue(RSA_size(rsa));

        // Defered free
        DEFER([&]() { T::RSA_free(rsa); });

        const auto encryptedLen = T::RSA_public_encrypt(
            input.length(), (const unsigned char*)input.data(), encryptedValue.data(), rsa, RSA_PKCS1_PADDING);

        if (encryptedLen < 0)
        {
            throw std::runtime_error("RSA encryption failed");
        }

        output = std::string(encryptedValue.begin(), encryptedValue.end());

        return encryptedLen;
    }

```